public/docs/ts/latest/guide/reactive-forms.jade

```diff
@@ -0,0 +1,879 @@
+
+:marked
+  Then add corresponding markup in `hero-detail.component.html` 
+  within the `form` element.
+
++makeExample('reactive-forms/ts/app/hero-detail-3.component.html', '','app/hero-detail.component.html')(format=".")
+
+.alert.is-helpful
+  :marked
+    *Reminder*: Ignore the many mentions of `form-group` and `form-control`in this markup.
+    Those are _bootstrap_ CSS classes that Angular itself ignores.
+    Pay attention to the `formGroupName` and `formControlName` attributes.
+    They are the Angular directives that bind the HTML controls to the 
+    Angular `FormGroup` and `FormControl` properties in the component class.
+
+.l-main-section
+a#grouping
+:marked
+  ### Group FormControls
+
+  This form is getting big and unwieldy.
+  You can make it more maintainable by grouping some of the related `FormControls` into another `FormGroup`.
+  The `street`, `city`, `state`, and `zip` are properties that would make a good _address_ `FormGroup`. 
+  
+  You used the `FormBuilder` to create one `FormGroup` in this component called `heroForm`.
+  Let that be the parent `FormGroup`.
+  Use `FormBuilder` again to create a child `FormGroup` that encapsulates the address controls;
+  assign the result to a new `address` property of the parent `FormGroup`.
++makeExample('reactive-forms/ts/app/hero-detail-versions.component.ts', 'v4','app/hero-detail.component.ts (excerpt)')(format=".")
+
+:marked
+  You’ve changed the structure of the form controls in the component class; 
+  you must make corresponding adjustments to the component template.
+  
+  In `hero-detail.component.html`, wrap the address-related `FormControls` in a `div`.
+  Add a `formGroupName` directive to the `div` and bind it to `"address"`.
+  That's the property of the _address_ child `FormGroup` within the parent `FormGroup` called `heroForm`.
+  
+  To make this change visually obvious, slip in an `<h4>` header near the top with the text, _Secret Lair_. 
+  The new _address_ HTML looks like this:
+
++makeExample('reactive-forms/ts/app/hero-detail-4.component.html', 'add-group','app/hero-detail.component.html (excerpt)')(format=".")
+
+:marked
+  After these changes, the JSON output in the browser shows the revised _form model_
+  with the nested address `FormGroup`: 
+
+figure.image-display
+  img(src="/resources/images/devguide/reactive-forms/address-group.png" width="400px" alt="JSON output")
+
+:marked
+  Great! You’ve made a group and you can see that the template 
+  and the form model are talking to one another.
+
+.l-main-section
+a#properties
+
+:marked
+  ## Inspect _FormControl_ Properties
+  At the moment, you're dumping the entire form model onto the page.
+  Sometimes you're interested only in the state of one particular `FormControl`.
+
+  You can inspect an individual `FormControl` within a form by extracting it with the `.get()` method.
+  Normally you'd do this _within_ the component class. 
```

Comment:
So we are expecting that the "normal" case will be defining individual properties for each control on the form in the component code? That has not been my experience so far.